Formula/node.rb

```diff
@@ -21,15 +21,14 @@ class Node < Formula
   end
 
   depends_on "pkg-config" => :build
-  depends_on "python@3.9" => :build
+  depends_on "python@3.10" => :build
   depends_on "brotli"
   depends_on "c-ares"
   depends_on "icu4c"
   depends_on "libnghttp2"
   depends_on "libuv"
   depends_on "openssl@1.1"
 
-  uses_from_macos "python", since: :catalina
```

Comment:
Removing this will break installs on Linux which don't already have a `python3` installation.

Reply:
Can we let `uses_from_macos "python"` determine which version of `python@3.x` will be used on Linux?

If already have `depends_on "python@3.10"[ => [:build][, :test]]`, then `"python"` means `python@3.10` on Linux rather than the aliased `python@3.9`.

Reply:
> Can we let uses_from_macos "python" determine which version of python@3.x will be used on Linux?

No, we can't. Or we need to change this in brew, but then we change it for all the other formulae.

IMHO we should make python@3.10 a runtime dependency for both mac and linux, and bump the revision.

Reply:
> IMHO we should make python@3.10 a runtime dependency for both mac and linux, and bump the revision.

We tried that previously and lots of users complained. I really don't think we should make `python@3.10` a dependency on Linux.

Reply:
Weird. Forgot about that. So it's a runtime dependency, and when you pin it to a brewed versions, users complain. What Python are they using instead on Linux?
Should we drop Python completely then?

Reply:
Well the Mac users complained. I think we need it on Linux since we can't safely expect a `python3` to be installed. But we can expect some `python3` to be there on macOS (and it seems pretty much any `python3` will do for `node`).

Reply:
> Or we need to change this in `brew`, but then we change it for all the other formulae.

That's exactly what I mean. `python` and `openssl` are the last two formulae both list in `ALLOWED_USES_FROM_MACOS_DEPS` and `versioned_keg_only_allowlist.json`.

https://github.com/Homebrew/brew/blob/40e3a47f12197955dbc892068b6aff05fa58f833/Library/Homebrew/rubocops/uses_from_macos.rb#L75-L90

https://github.com/Homebrew/homebrew-core/blob/93ad38a33b36e2f72abbf1a88825a35fadbc9a76/audit_exceptions/versioned_keg_only_allowlist.json#L1-L31